finesse/hardware/plugins/time/ntp_time.py

```diff
@@ -49,23 +54,33 @@ def __init__(
         self._ntp_port = ntp_port
         self._ntp_timeout = ntp_timeout
 
-    def close(self) -> None:
-        """Close the connection to the device."""
-
-    def get_time(self) -> float:
-        """Get the current time.
+        # Set up time offset polling.
+        self.poll_time_offset()
+        self._poll_timer = QTimer()
+        self._poll_timer.timeout.connect(self.poll_time_offset)
+        self._poll_timer.start(int(ntp_poll_interval * 1000))
 
-        Returns:
-            The current time in seconds since the epoch.
-        """
+    def poll_time_offset(self) -> None:
+        """Query the NTP server for the current time offset."""
         try:
-            response = self._client.request(
+            self._response = self._client.request(
                 self._ntp_host,
                 version=self._ntp_version,
                 port=self._ntp_port,
                 timeout=self._ntp_timeout,
             )
         except Exception as e:
-            raise NTPTimeError(f"Error querying NTP server: {e}")
+            error = NTPTimeError(e)
+            self.send_error_message(error)
+
+    def get_time_offset(self) -> float:
+        """Get the current time offset.
 
-        return response.dest_time + response.offset
+        Returns:
+            A float representing the current time offset.
+        """
+        return self._response.offset
+
+    def close(self) -> None:
+        """Close the device."""
+        self._poll_timer.stop()
```

Comment:
You also need to explicitly call the parent class's `close()` because there is some `Device`-specific cleanup that needs to happen:

```suggestion
        self._poll_timer.stop()
        super().close()
```